content-src/components/Snippet/Snippet.js

```diff
@@ -0,0 +1,57 @@
+const React = require("react");
+const classNames = require("classnames");
+
+const Snippet = React.createClass({
+  onClose(e) {
+    e.preventDefault();
+    this.props.setVisibility(false);
+  },
+  render() {
+    const {enabled, visible, description, title, image} = this.props;
```

Comment:
As mentioned before, I think your suggestion of {isThereASnippet && <Snippet ... /.>} is just right, which should get rid of the need for one of these props.

You might (I'm not sure) also be able to get rid of the need for the other using [ReactCSSTransitionGroup](https://facebook.github.io/react/docs/animation.html#animating-one-or-zero-items).  That may or may not be worth the effort this time around.